scanner/services/indexer.go

```diff
@@ -8,14 +8,33 @@ import (
 	"github.com/quay/claircore"
 	"github.com/quay/zlog"
 	v4 "github.com/stackrox/rox/generated/internalapi/scanner/v4"
+	"github.com/stackrox/rox/pkg/buildinfo"
 	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/pkg/grpc/authz"
 	"github.com/stackrox/rox/pkg/grpc/authz/allow"
+	"github.com/stackrox/rox/pkg/grpc/authz/idcheck"
+	"github.com/stackrox/rox/pkg/grpc/authz/or"
+	"github.com/stackrox/rox/pkg/grpc/authz/perrpc"
 	"github.com/stackrox/rox/scanner/indexer"
 	"github.com/stackrox/rox/scanner/mappers"
 	"github.com/stackrox/rox/scanner/services/validators"
 	"google.golang.org/grpc"
 )
 
+var indexerAuth = func() authz.Authorizer {
+	if buildinfo.ReleaseBuild {
+		return perrpc.FromMap(map[authz.Authorizer][]string{
+			or.SensorOr(idcheck.CentralOnly()): {
```

Comment:
IIRC the indexer part will be used on secured clusters for the local scanning workflow, whereas the matching part will stay (for the time being) Central specific.

While currently it's not possible to constrain access here to a _specific_ sensor, i.e. the sensor associated with the local scanner deployment. It's theoretically possible that a Sensor from a different cluster can use scanners from other clusters to send scan requests to.
But since the scanner endpoints aren't supposed to be exposed (unless this changes), think this makes the point moot (also probably not that big of a deal if it would work since indexing should be the same across every cluster?).

Reply:
yeah I don't think it's a big deal if a Sensor in another cluster manages to reach out. Would be weird, and I think we'd have other problems, as we definitely don't want to do that (at this time)